### PR TITLE
replaced the deprecated pedantic package with the new lints package

### PR DESCRIPTION
**Pull request by danger-ahead**
fixed analyser issues:
- Invalid use of a private type in a public API.
- Constructors in '@immutable' classes should be declared as 'const'

### CODE REVIEW

**Comment by danger-ahead** [comment]
@Miiite @nilsreichardt I have pushed minor code changes to fix the flutter analyzer issues and remove the deprecated pedantic package. Please review.

File: pubspec.yaml [hunk 1]
```diff
@@ -19,4 +19,4 @@ dependencies:
 dev_dependencies:
   flutter_test:
     sdk: flutter
-  pedantic: ^1.7.0
+  lints: ^2.0.1
```
Comment (Miiite) [inline pubspec.yaml:22]: I think it should be `flutter_lint` here ?

File: analysis_options.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-include: package:pedantic/analysis_options.yaml
+include: package:lints/recommended.yaml
```
Comment (Miiite) [inline analysis_options.yaml:1]: Based on below package suggestion, we should change that to:

```suggestion
include: package:flutter_lints/flutter.yaml
```

**Review by Miiite — CHANGES_REQUESTED** [review]

File: pubspec.yaml (hunk 1 above)
Comment (danger-ahead) [reply pubspec.yaml:22]: yes. thanks. I overlooked.

**Review by Miiite — APPROVED** [review]

**Pull request merged**